lib/kitchen/provisioner/chef/common_sandbox.rb

```diff
@@ -278,6 +295,27 @@ def prepare_json
           end
         end
 
+        def update_dna_for_policyfile
+          if !config[:run_list].nil? && !config[:run_list].empty?
+            warn("You must set your run_list in your policyfile instead of "\
+                 "kitchen config. The run_list your config will be ignored.")
+            warn("Ignored run_list: #{config[:run_list].inspect}")
+          end
+          policylock = policyfile.gsub(/\.rb\Z/, ".lock.json")
+          policy_name = JSON.parse(IO.read(policylock))["name"]
```

Comment:
Good catch. I thought that `chef export` would get called first, but actually it's not and indeed that case will fail with `ENOENT`.